project/Build.scala

```diff
@@ -1490,14 +1490,20 @@ object Build {
             generateDocumentation(classDirectory.in(Compile).value.getAbsolutePath, "scala3doc", "self", "-p documentation")
           }.value,
           generateScala3Documentation := Def.taskDyn {
+            // ensure all projects we're documenting are compiled
+            (`scala3-interfaces`/Compile/compile).value
+            (`tasty-core`/Compile/compile).value
+            (`scala3-library`/Compile/compile).value
+            // TODO we can't load stdlib from Tasty
+            // (`stdlib-bootstrapped`/Compile/compile).value
+
             val dottyJars = Seq(
-              // All projects below will be used to generated documentation for Scala 3
-              classDirectory.in(`scala3-interfaces`).in(Compile).value,
```

Comment:
Try using `products` instead of `classDirectory` to still gt the directories back but after they're done compiling